src/client/content_mapblock.cpp

```diff
@@ -366,13 +366,22 @@ void MapblockMeshGenerator::generateCuboidTextureCoords(const aabb3f &box, f32 *
 void MapblockMeshGenerator::drawAutoLightedCuboid(aabb3f box, const f32 *txc,
 	TileSpec *tiles, int tile_count)
 {
+	bool scale = std::fabs(f->visual_scale - 1.0) > 1e-3;
 	f32 texture_coord_buf[24];
 	f32 dx1 = box.MinEdge.X;
 	f32 dy1 = box.MinEdge.Y;
 	f32 dz1 = box.MinEdge.Z;
 	f32 dx2 = box.MaxEdge.X;
 	f32 dy2 = box.MaxEdge.Y;
 	f32 dz2 = box.MaxEdge.Z;
+	if (scale) {
+		if (!txc) { // generate texture coords before scaling
+			generateCuboidTextureCoords(box, texture_coord_buf);
+			txc = texture_coord_buf;
```

Comment:
I don't think this special casing is needed, just move the `if (!txc) {` before the `if (scale) {`

Reply:
That would break world-aligned textures

Reply:
Hm, okay.